Lib/test/test_capi/test_opt.py

```diff
@@ -1923,6 +1923,19 @@ def testfunc(n):
         self.assertNotIn("_GUARD_TOS_INT", uops)
         self.assertIn("_CALL_LEN", uops)
 
+    def test_get_len(self):
+        def testfunc(n):
+            a = [1, 2, 3, 4]
+            for _ in range(n):
+                match a:
+                    case [1,2]:
+                        pass
+        _, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
```

Comment:
If we change the test slightly, we can make sure that the expected behavior is still occurring by checking the number of matches:

```suggestion
            x = 0
            a = [1, 2, 3, 4]
            for _ in range(n):
                match a:
                    case [_, _, _, _]:
                        x += 1
        re, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
        self.assertEqual(res, TIER2_THRESHOLD)
```